impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java

```diff
@@ -329,6 +358,49 @@ public ClientBuilder setOrgUrl(String baseUrl) {
         return this;
     }
 
+    @Override
+    public ClientBuilder setAuthorizationMode(String authorizationMode) {
+        if (Strings.isEmpty(authorizationMode)) {
+            throw new IllegalArgumentException("Invalid AuthorizationMode");
+        }
+        this.clientConfig.setAuthorizationMode(authorizationMode);
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setScopes(List<String> scopes) {
+        if (scopes == null || scopes.size() == 0) {
```

Comment:
Don't you have to validate this only when using OAuth2?

Reply:
💯

Reply:
True. will factor that in.